server/controllers/cohortController.js

```diff
@@ -1,20 +1,50 @@
 import db from "../db";
 export const getAllCohorts = async (req, res) => {
-    try {
-      const q = "SELECT * FROM cohorts";
-      const result = await db.query(q);
-      const cohorts = result.rows;
-      res.status(200).json({ cohorts });
-    } catch (error) {
-      console.error("Error retrieving cohorts:", error);
-      res.status(500).send("Failed, Internal Server Error.");
-    }
-  };
+  try {
+    const q = "SELECT * FROM cohorts";
+    const result = await db.query(q);
+    const cohorts = result.rows;
+    res.status(200).json({ cohorts });
+  } catch (error) {
+    console.error("Error retrieving cohorts:", error);
+    res.status(500).send("Failed, Internal Server Error.");
+  }
+};
 
-export const getCohort = (req, res) => {
-    res.status(200).json("you can write your get function to get one cohort here");
+export const getCohort = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const q = "SELECT * FROM cohorts WHERE id = $1";
+    const result = await db.query(q, [id]);
+    const targetedCohort = result.rows[0];
+    if (targetedCohort) {
+      console.log("Cohort found:", targetedCohort);
+      res.status(200).json({ cohort: targetedCohort, message: "Cohort retrieved successfully" });
+    } else {
+      console.error("Cohort not found.");
+      res.status(404).json({ error: "Cohort not found." });
+    }
+  } catch (error) {
+    console.error("Error retrieving cohort:", error);
+    res.status(500).send("Failed to retrieve cohort.");
+  }
 };
 
-export const createCohort = (req, res) => {
-    res.status(200).json("you can write your post function to create new cohort here");
-};
\ No newline at end of file
+export const createCohort = async (req, res) => {
+  try {
+    const { name, start_date } = req.body; //Destructuring
+    if (!name || !start_date) {
+      return res.status(400).json({ error: "Please provide both Name and start_date." });
+    }
```

Comment:
maybe we can get this error in front end without, so we can first be sure in front end about Input then fetch Api (I am not sure how much this idea is correct)

Reply:
we can get errors in frontend too, but the good way to handle errors is to be on server-side, but we will discuss it

Reply:
Yeah it is good to be in both